src/main/scala/org/broadinstitute/dsde/firecloud/service/ExportEntitiesByTypeActor.scala

```diff
@@ -94,19 +93,30 @@ class ExportEntitiesByTypeActor(rawlsDAO: RawlsDAO,
     *
     * Handle exceptions directly by completing the request.
     */
-  def streamEntities(): Future[Unit] = {
+  def streamEntities(): Future[HttpResponse] = {
+    val keepAlive = Connection("Keep-Alive")
+
     entityTypeMetadata flatMap { metadata =>
       val entityQueries = getEntityQueries(metadata, entityType)
       if (modelSchema.isCollectionType(entityType)) {
-        streamCollectionType(entityQueries, metadata)
+        val contentType = ContentTypes.`application/octet-stream`
+        val fileName = entityType + ".zip"
+        val disposition = `Content-Disposition`.apply(ContentDispositionTypes.attachment, Map("filename" -> fileName))
+
+        streamCollectionType(entityQueries, metadata).map { source =>
+          HttpResponse(entity = HttpEntity.fromFile(contentType, source.toJava), headers = List(keepAlive, disposition))
+        }
       } else {
         val headers = TSVFormatter.makeEntityHeaders(entityType, metadata.attributeNames, attributeNames)
-        streamSingularType(entityQueries, metadata, headers)
-      }.recover { case t: Throwable =>
-        // Stream exceptions have to be handled by directly closing out the RequestContext responder
-        handleStreamException(t)
+
+        val contentType = ContentType(MediaTypes.`text/tab-separated-values`, HttpCharsets.`UTF-8`)
+        val fileName = entityType + ".tsv"
+        val disposition = `Content-Disposition`.apply(ContentDispositionTypes.attachment, Map("filename" -> fileName))
+
+        streamSingularType(entityQueries, metadata, headers).map { source =>
+          HttpResponse(entity = HttpEntity.fromFile(contentType, source.toJava), headers = List(keepAlive, disposition))
```

Comment:
`fromFile` should utilize chunking (with the default chunk size) but admittedly i haven't looked up what that default chunk size is

Reply:
default is 8192 I believe, specified in `akka.stream.scaladsl.FileIO.fromPath`